sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Catalog.scala

```diff
@@ -34,6 +34,12 @@ trait Catalog {
       tableIdentifier: Seq[String],
       alias: Option[String] = None): LogicalPlan
 
+  /**
+   * Returns names and flags indicating if a table is temporary or not of all tables in the
+   * database identified by `databaseIdentifier`.
+   */
+  def getTables(databaseIdentifier: Seq[String]): Seq[(String, Boolean)]
```

Comment:
also you should explain in the javadoc what the returned tuples mean (boolean field?)